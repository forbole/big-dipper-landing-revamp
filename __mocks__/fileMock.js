// __mocks__/fileMock.js
module.exports = {
  blurDataURL: 'data:image/png;base64,imagedata',
  height: 24,
  src: '/img.jpg',
  width: 24,
}