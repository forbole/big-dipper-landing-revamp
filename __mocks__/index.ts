export { default as MockTheme } from "./theme";
