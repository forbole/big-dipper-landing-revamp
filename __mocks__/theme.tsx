/* eslint-disable */
import { ThemeProvider } from '@mui/material/styles';
import { ReactNode } from 'react';
import { darkTheme } from '@/src/styles/theme';

/**
 * Theme mocker to handle custom keys
 */
const MockTheme = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={darkTheme}>{children}</ThemeProvider>;
};

export default MockTheme;
