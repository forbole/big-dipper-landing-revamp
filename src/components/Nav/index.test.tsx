/* eslint-disable react/display-name */
import { MockTheme } from "@/__mocks__";
import Box from "@mui/material/Box";
import { render } from "@testing-library/react";

import Component from ".";

jest.mock("./components/MenuDesktop", () => () => (
  <Box data-testid="MenuDesktop" />
));

jest.mock("./components/MenuMobile", () => () => (
  <Box data-testid="MenuMobile" />
));

// ==================================
// unit tests
// ==================================
describe("Nav", () => {
  it("matches snapshot", async () => {
    const { container } = render(
      <MockTheme>
        <Component />
      </MockTheme>,
    );

    // expect(screen.getByText(
    //   /Home/i,
    // )).toBeInTheDocument();
    // expect(screen.getByText(
    //   /About/i,
    // )).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
