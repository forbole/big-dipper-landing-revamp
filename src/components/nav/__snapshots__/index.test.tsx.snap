// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav matches snapshot 1`] = `
.emotion-0 {
  margin-left: auto;
  margin-right: auto;
  max-width: 1024px;
  position: relative;
  width: 100%;
}

.emotion-0>button {
  height: 60px;
  position: absolute;
  padding: 40px;
  z-index: 1;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 79px;
  padding: 16px;
}

@media (min-width:768px) {
  .emotion-1 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    justify-content: space-between;
  }
}

.emotion-2 {
  height: 36px;
  display: block;
}

@media (min-width:768px) {
  .emotion-2 {
    margin: 8px;
  }
}

.emotion-3 {
  background: unset;
  border: unset;
  height: 60px;
  width: 60px;
  position: absolute;
  right: 0px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  z-index: 2;
}

.emotion-3 .bar1,
.emotion-3 .bar2,
.emotion-3 .bar3 {
  width: 24px;
  border: 1px solid #FFFFFF;
  margin: 6px 0;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  border-radius: 25px;
}

.emotion-3 .hamburger-X .bar2 {
  opacity: 0;
}

.emotion-3 .hamburger-X .bar1 {
  -webkit-transform: rotate(-45deg) translate(-4.5px, 4.5px);
  -moz-transform: rotate(-45deg) translate(-4.5px, 4.5px);
  -ms-transform: rotate(-45deg) translate(-4.5px, 4.5px);
  transform: rotate(-45deg) translate(-4.5px, 4.5px);
}

.emotion-3 .hamburger-X .bar3 {
  -webkit-transform: rotate(45deg) translate(-6.5px, -6.5px);
  -moz-transform: rotate(45deg) translate(-6.5px, -6.5px);
  -ms-transform: rotate(45deg) translate(-6.5px, -6.5px);
  transform: rotate(45deg) translate(-6.5px, -6.5px);
}

@media (min-width:768px) {
  .emotion-3 {
    display: none;
  }
}

.emotion-4 {
  display: none;
  width: auto;
  -webkit-box-pack: right;
  -ms-flex-pack: right;
  -webkit-justify-content: right;
  justify-content: right;
}

@media (min-width:768px) {
  .emotion-4 {
    display: -webkit-inline-box;
    display: -webkit-inline-flex;
    display: -ms-inline-flexbox;
    display: inline-flex;
  }
}

.emotion-4>.MuiList-root>.MuiListItemButton-root:first-of-type {
  display: none!important;
}

.emotion-4 .MuiList-root {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 .MuiButtonBase-root {
  padding: 8px 24px;
  border-radius: 24px;
}

.emotion-4 .submenu,
.emotion-4 .active {
  background: transparent none!important;
}

.emotion-4 .active.submenu {
  background: linear-gradient(103.18deg, #F6504B 20.77%, #DC001A 118.69%)!important;
}

.emotion-4 .submenu-box {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
}

.emotion-4 .submenu-container {
  position: absolute;
  top: 100%;
  width: 100%;
  margin-top: 16px;
  background: #3c52af none;
  border-radius: 16px;
  z-index: 1;
}

.emotion-4 .submenu-container .MuiButtonBase-root {
  padding: 16px 24px;
}

.emotion-5 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-width: 0;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding-left: 16px;
  padding-right: 16px;
  text-align: left;
  padding: 8px 32px;
}

.emotion-6::-moz-focus-inner {
  border-style: none;
}

.emotion-6.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-6 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-6:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(255, 255, 255, 0.08);
}

@media (hover: none) {
  .emotion-6:hover {
    background-color: transparent;
  }
}

.emotion-6.Mui-selected {
  background-color: rgba(144, 202, 249, 0.16);
}

.emotion-6.Mui-selected.Mui-focusVisible {
  background-color: rgba(144, 202, 249, 0.28);
}

.emotion-6.Mui-selected:hover {
  background-color: rgba(144, 202, 249, 0.24);
}

@media (hover: none) {
  .emotion-6.Mui-selected:hover {
    background-color: rgba(144, 202, 249, 0.16);
  }
}

.emotion-6.Mui-focusVisible {
  background-color: rgba(255, 255, 255, 0.12);
}

.emotion-6.Mui-disabled {
  opacity: 0.38;
}

.emotion-6.MuiListItem-gutters {
  padding: 16px 20px;
}

.emotion-6 .MuiListItemText-root {
  color: #fff;
  font-weight: 900;
}

.emotion-6.active {
  background: linear-gradient(103.18deg, #F6504B 20.77%, #DC001A 118.69%);
}

.emotion-6.submenu {
  background: #3c52af none;
}

.emotion-7 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 4px;
  margin-bottom: 4px;
}

.emotion-8 {
  margin: 0;
  font-size: 1rem;
  white-space: pre-wrap;
  font-family: SFPro,Arial,sans-serif;
  font-weight: 400;
  line-height: 1.5;
  display: block;
}

.emotion-9 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <div>
    <div
      class="emotion-0"
    >
      <div
        class="wrapper"
      >
        <span
          class="logo"
          height="36"
        />
        <button
          class="emotion-1"
          type="button"
        >
          <div
            class="hamburger"
          >
            <div
              class="bar1"
            />
            <div
              class="bar2"
            />
            <div
              class="bar3"
            />
          </div>
        </button>
      </div>
      <div
        class="MuiBox-root emotion-4"
      >
        <ul
          class="MuiList-root MuiList-padding emotion-5"
          role="menu"
          tabindex="-1"
        >
          <a
            class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters  emotion-6"
            href="/"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root emotion-7"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
              >
                home
              </span>
            </div>
            <span
              class="MuiTouchRipple-root emotion-9"
            />
          </a>
          <a
            class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters  emotion-6"
            href="/all-networks"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root emotion-7"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
              >
                allNetworks
              </span>
            </div>
            <span
              class="MuiTouchRipple-root emotion-9"
            />
          </a>
          <div
            class="submenu-box MuiBox-root emotion-14"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters submenu emotion-6"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root emotion-7"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
                >
                  helpsAndSupport
                </span>
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-18"
                data-testid="KeyboardArrowDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root emotion-9"
              />
            </div>
            <div
              class="submenu-container MuiBox-root emotion-14"
            />
          </div>
          <a
            class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters  emotion-6"
            href="/about"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root emotion-7"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
              >
                about
              </span>
            </div>
            <span
              class="MuiTouchRipple-root emotion-9"
            />
          </a>
          <a
            class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters  emotion-6"
            href="https://docs.bigdipper.live/"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root emotion-7"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-8"
              >
                docs
              </span>
            </div>
            <span
              class="MuiTouchRipple-root emotion-9"
            />
          </a>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
